docs/examples/101/webapp-privateendpoint-vnet-injection/main.bicep

```diff
@@ -0,0 +1,187 @@
+
+resource virtualNetwork 'Microsoft.Network/virtualNetworks@2020-06-01' = {
+  name: virtualNetworkName
+  location: location
+  properties: {
+    addressSpace: {
+      addressPrefixes: [
+        virtualNetwork_CIDR
+      ]
+    }
+  }
+}
+
+resource subnet1 'Microsoft.Network/virtualNetworks/subnets@2020-06-01' = {
+  name: '${virtualNetwork.name}/${subnet1Name}'
+  properties: {
+    addressPrefix: subnet1_CIDR
+    privateEndpointNetworkPolicies: 'Disabled'
+  }
+}
+
+resource subnet2 'Microsoft.Network/virtualNetworks/subnets@2020-06-01' = {
+  name: '${virtualNetwork.name}/${subnet2Name}'
+  properties: {
+    addressPrefix: subnet2_CIDR
+    delegations: [
+      {
+        name: 'delegation'
+        properties: {
+          serviceName: 'Microsoft.Web/serverfarms'
+        }
+      }
+    ]
+    privateEndpointNetworkPolicies: 'Enabled'
+  }
+  dependsOn: [
+    subnet1
+  ]
+}
+
+resource serverFarm 'Microsoft.Web/serverfarms@2020-06-01' = {
+  name: serverFarmName
+  location: location
+  sku: {
+    name: skuName
+    tier: SKU_tier
+    size: skuName
+    family: skuName
+    capacity: 1
+  }
+  kind: 'app'
+}
+
+resource webApp1 'Microsoft.Web/sites@2020-06-01' = {
+  name: site1_Name
+  location: location
+  kind: 'app'
+  properties: {
+    serverFarmId: serverFarm.id
+  }
+}
+
+resource webApp2 'Microsoft.Web/sites@2020-06-01' = {
+  name: site2_Name
+  location: location
+  kind: 'app'
+  properties: {
+    serverFarmId: serverFarm.id
+  }
+}
+resource webApp2AppSettings 'Microsoft.Web/sites/config@2020-06-01' = {
+  name: '${webApp2.name}/appsettings'
+  properties: {
+    'WEBSITE_DNS_SERVER': '168.63.129.16'
+    'WEBSITE_VNET_ROUTE_ALL': '1'
+  }
+}
+
+resource webApp1Config 'Microsoft.Web/sites/config@2020-06-01' = {
+  name: '${webApp1.name}/web'
+  properties: {
+    ftpsState: 'AllAllowed'
+  }
+}
+
+resource webApp2Config 'Microsoft.Web/sites/config@2020-06-01' = {
+  name: '${webApp2.name}/web'
+  properties: {
+    ftpsState: 'AllAllowed'
+  }
+}
+
+resource webApp1Binding 'Microsoft.Web/sites/hostNameBindings@2019-08-01' = {
+  name: '${webApp1.name}/${webApp1.name}${webapp_dns_name}'
+  properties: {
+    siteName: webApp1.name
+    hostNameType: 'Verified'
+  }
+}
+
+resource webApp2Binding 'Microsoft.Web/sites/hostNameBindings@2019-08-01' = {
+  name: '${webApp2.name}/${webApp2.name}${webapp_dns_name}'
+  properties: {
+    siteName: webApp2.name
+    hostNameType: 'Verified'
+  }
+}
+
+resource webApp2NetworkConfig 'Microsoft.Web/sites/networkConfig@2020-06-01' = {
+  name: '${webApp2.name}/VirtualNetwork'
+  properties: {
+    subnetResourceId: resourceId('Microsoft.Network/virtualNetworks/subnets', virtualNetwork.name, subnet2Name)
+  }
+}
+
+resource privateEndpoint 'Microsoft.Network/privateEndpoints@2020-06-01' = {
+  name: 'PrivateEndpoint1'
+  location: location
+  properties: {
+    subnet: {
+      id: resourceId('Microsoft.Network/virtualNetworks/subnets', virtualNetwork.name, subnet1Name)
```

Comment:
since `subnet1` is declared as a child resource, you should be able to do `subnet1.id`